dotcom-rendering/src/web/components/HeaderTopBarSearchCapi.tsx

```diff
@@ -0,0 +1,211 @@
+import { css } from '@emotion/react';
+import {
+	brand,
+	brandAlt,
+	from,
+	neutral,
+	textSans,
+} from '@guardian/source-foundations';
+import { useEffect, useRef, useState } from 'react';
+import SearchIcon from '../../static/icons/search.svg';
+import { getZIndex } from '../lib/getZIndex';
+import { useApi } from '../lib/useApi';
+
+const searchLinkStyles = css`
+	${textSans.medium({ fontWeight: 'bold' })};
+	color: ${neutral[100]};
+	transition: color 80ms ease-out;
+	text-decoration: none;
+	padding: 7px 0;
+	background: none;
+	border: none;
+
+	${from.tablet} {
+		padding: 7px 10px 7px 5px;
+	}
+
+	:hover,
+	:focus {
+		text-decoration: underline;
+	}
+
+	svg {
+		fill: currentColor;
+		float: left;
+		height: 18px;
+		width: 18px;
+		margin: 3px 4px 0 0;
+	}
+	${getZIndex('searchHeaderLink')}
+`;
+
+const linkTablet = css`
+	display: none;
+	position: relative;
+
+	:before {
+		content: '';
+		border-left: 1px solid ${brand[600]};
+		height: 24px;
+	}
+
+	${from.desktop} {
+		display: flex;
+	}
+`;
+
+export const HeaderTopBarSearchCapi = () => {
+	const [isSearchVisible, setIsSearchVisible] = useState(false);
+	const [q, setQ] = useState<string>();
+	const qRef = useRef<HTMLInputElement>(null);
+	const { data } = useApi<{
+		response: { results: { webTitle: string; webUrl: string }[] };
+	}>(
```

Comment:
If we wanted to add a litte bit of parsing, here’s a [zod implementation](https://github.com/mxdvl/dcr-json-comparison/blob/164d4790c13f397d412bb6ef9efc064521a65afe/get-articles.ts#L24-L45):

```ts
import { string, literal, object, array } from "zod";


const status = literal("ok");

const resultsSchema = object({
  response: object({
    status,
    results: array(
      object({
        webTitle: string(),
        webUrl: string(),
      })
    ),
  }),
});
```

Reply:
`zod` is not available, and given that this will run on this client, using it would mean having to put it in the client bundle, which feels not-so-performant 💨

Reply:
Haha I just tried to coerce you into bringing into the repo by “mistake” 🤣